packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +105,132 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// This class stores the information of an active selection under
+/// a [Selectable] or [SelectionHandler], including the start and
+/// end positions of a selection relative to the content, as well
+/// as an identifier to identify the owner of the content, any
+/// children selection ranges that may be contained within the active
+/// selection, and the content itself.
+///
+/// [SelectionArea] and [SelectableRegion] provide access to this
+/// information through the [SelectionListener] passed to their
+/// [SelectionListener.onSelectionChanged] member.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides the [SelectedContentRange]s representing
+///     the active selection for a given subtree contained under a [SelectionArea]
+///     or [SelectableRegion].
+abstract class SelectedContentRange<T extends Object> {
```

Comment:
I made the class concrete and made startOffset and endOffset fields. I think my main thought was by making it abstract it would be more flexible, but as stated in reviews it has made it more verbose.